### DISCUSSION TITLE
No Output from CreateModule (Scripting.Python)

### DISCUSSION THREAD

**Discussion by jaroducsb** [bonsai-rx/bonsai#1998] [Q&A]
Hello!

I'm attempting to utilize the Python Scripting package to use some external libraries in Bonsai. However, I haven't been able to get an output from CreateModule when pasting the code in from [the documentation](https://bonsai-rx.org/python-scripting/articles/modules.html). It runs without errors, but Set nor Eval seem to return anything in their visualizers. I tried to set it up on my own (KeyDown instead of Timer) and there also weren't any outputs. For reference, the CreateRuntime code worked flawlessly and the visualizers were showing the expected outputs in both the examples and my own tinkering. 

Speaking of which, I was using CreateRuntime successfully until Bonsai crashed and I started getting the Delegates error seen in [this comment](https://github.com/orgs/bonsai-rx/discussions/1926#discussioncomment-10081722). In case anyone else runs into this issue, adding the Python kernel as a PATH environment variable seems to have remedied that and I haven't run into any issues since. 

I'd appreciate any insight on the former issue though!

**Comment by ncguilbeault** [comment]
Can you try switching the `ResourceSubject` with a `BehaviorSubject` and see if it works?

**Reply by jaroducsb** [reply]
Thank you for the quick reply! That unfortunately didn't change anything. I tried all of the subjects just in case, none of them yielded anything. Any other ideas? I am also running into the Delegates error for CreateRuntime again so my solution didn't seem to be permanent (EDIT: activating the virtual environment seemed to fix this again).

Update: Embarrassingly thought CreateRuntime and CreateModule were mutually exclusive when we actually need both. Works as expected now, thank you again for the response!

**Reply by jaroducsb** [reply]
Apologies if this should be a separate thread, but do you happen to know if it's possible to use the output of one of these Python modules for use within Bonsai? As a simple example, could I use Get to grab a variable from an external script and plug that output into a Float?

**Reply by ncguilbeault** [reply]
Let's open a separate discussion for this and continue from there
